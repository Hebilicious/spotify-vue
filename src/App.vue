<script setup lang="ts">
import { onMounted } from "vue";
import { RouterView } from "vue-router";

import NavBar from "@/components/NavBar.vue";
import PlayerBar from "@/components/PlayerBar.vue";

import usePlayListStore from "@/stores/playlist";

onMounted(() => usePlayListStore().loadPlayLists());

</script>

<template>
  <NavBar />
  <div class="main-view">
    <RouterView></RouterView>
  </div>
  <PlayerBar />
</template>

<style>
@import "./assets/base.css";

#app {
  font-weight: normal;
  height: 100%;
  display: grid;
  grid-template-areas:
    "nav-bar main-view"
    "now-playing-bar now-playing-bar";
  grid-template-columns: auto 1fr;
  grid-template-rows: 1fr auto;
}

.main-view {
  display: flex;
  grid-area: main-view;
  flex: 6;
  flex-direction: column;
}
</style>
