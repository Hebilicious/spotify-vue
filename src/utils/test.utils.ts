import { VueWrapper } from '@vue/test-utils';
import { expect, it } from 'vitest';

export const testSnapShot = (wrapperGetter: () => VueWrapper) => {
  it('Matches the snapshot', () => {
    expect(wrapperGetter().html()).toMatchSnapshot();
  });
};
