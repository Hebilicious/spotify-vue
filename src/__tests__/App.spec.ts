import { shallowMount, VueWrapper } from '@vue/test-utils';
import { afterEach, beforeEach, describe, fn } from 'vitest';
import { createTestingPinia } from '@pinia/testing';

import App from '@/App.vue';

import { testSnapShot } from '@/utils/test.utils';

describe('App', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = shallowMount(App, { global: { plugins: [createTestingPinia({ createSpy: () => fn() })] } });
  });

  afterEach(() => {
    wrapper.unmount();
  });

  testSnapShot(() => wrapper);
});
