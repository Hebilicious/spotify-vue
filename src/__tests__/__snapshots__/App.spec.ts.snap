// Vitest Snapshot v1

exports[`App > Matches the snapshot 1`] = `
"<nav-bar-stub></nav-bar-stub>
<div class=\\"main-view\\">
  <router-view-stub name=\\"default\\"></router-view-stub>
</div>
<player-bar-stub></player-bar-stub>"
`;
